Initial commit for Ansible based Satellite6 Install and Configure.

kbidarkar opened the pull request:
* Added a partition-disk role
* Added playbooks for sat63 and sat64

kbidarkar committed:
Initial commit for Ansible based Satellite6 Install and Configure.

* Added a partition-disk role
* Added playbooks for sat63 and sat64

Signed-off-by: Kedar Bidarkar <kbidarka@redhat.com>

kbidarkar commented:
[First run for the host: sat64-rhel7] Below are the results for the partition-disk role while using the sat64 playbook.

```
 git_world  ansible-satellite6  ansible-playbook -i inventory playbooks/install/satellite_64.yml

PLAY [sat64] *****************************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *******************************************************************************************************************************************************************************************************************************************************************
ok: [sat64-rhel7]

TASK [partition-disk : include_tasks] ****************************************************************************************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/remove_home_partition.yaml for sat64-rhel7

TASK [partition-disk : Unmount /home partition] ******************************************************************************************************************************************************************************************************************************************
changed: [sat64-rhel7]

TASK [partition-disk : Remove /home entry from /etc/fstab] *******************************************************************************************************************************************************************************************************************************
changed: [sat64-rhel7]

TASK [partition-disk : Remove /home Logical Volume] **************************************************************************************************************************************************************************************************************************************
changed: [sat64-rhel7] => (item={'key': 'home', 'value': {'size_g': '406.88', 'vg': 'rhel_cloud-qe-22'}})
skipping: [sat64-rhel7] => (item={'key': 'root', 'value': {'size_g': '50.00', 'vg': 'rhel_cloud-qe-22'}}) 
skipping: [sat64-rhel7] => (item={'key': 'swap', 'value': {'size_g': '7.88', 'vg': 'rhel_cloud-qe-22'}}) 

TASK [partition-disk : include_tasks] ****************************************************************************************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/extend_root_partition.yaml for sat64-rhel7

TASK [partition-disk : Extend root partition LV to all available VG space] ***************************************************************************************************************************************************************************************************************
skipping: [sat64-rhel7] => (item={'key': 'home', 'value': {'size_g': '406.88', 'vg': 'rhel_cloud-qe-22'}}) 
changed: [sat64-rhel7] => (item={'key': 'root', 'value': {'size_g': '50.00', 'vg': 'rhel_cloud-qe-22'}})
skipping: [sat64-rhel7] => (item={'key': 'swap', 'value': {'size_g': '7.88', 'vg': 'rhel_cloud-qe-22'}}) 

TASK [partition-disk : Resize the root filesystem] ***************************************************************************************************************************************************************************************************************************************
skipping: [sat64-rhel7] => (item={'key': 'home', 'value': {'size_g': '406.88', 'vg': 'rhel_cloud-qe-22'}}) 
changed: [sat64-rhel7] => (item={'key': 'root', 'value': {'size_g': '50.00', 'vg': 'rhel_cloud-qe-22'}})
skipping: [sat64-rhel7] => (item={'key': 'swap', 'value': {'size_g': '7.88', 'vg': 'rhel_cloud-qe-22'}}) 

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************************
sat64-rhel7                : ok=8    changed=5    unreachable=0    failed=0   
```

[Re-run for the host sat63-rhel7]Also, tried re-running the partition-disk role using sat63 playbook.

```
 git_world  ansible-satellite6  ansible-playbook -i inventory playbooks/install/satellite_63.yml

PLAY [sat63] *****************************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] *******************************************************************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : include_tasks] ****************************************************************************************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/remove_home_partition.yaml for sat63-rhel7

TASK [partition-disk : Unmount /home partition] ******************************************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : Remove /home entry from /etc/fstab] *******************************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7]

TASK [partition-disk : Remove /home Logical Volume] **************************************************************************************************************************************************************************************************************************************
skipping: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '262.68', 'vg': 'rhel_hp-dl380-gen9-5'}}) 
skipping: [sat63-rhel7] => (item={'key': 'swap', 'value': {'size_g': '15.69', 'vg': 'rhel_hp-dl380-gen9-5'}}) 

TASK [partition-disk : include_tasks] ****************************************************************************************************************************************************************************************************************************************************
included: /home/kbidarka/git_world/ansible-satellite6/roles/partition-disk/tasks/extend_root_partition.yaml for sat63-rhel7

TASK [partition-disk : Extend root partition LV to all available VG space] ***************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '262.68', 'vg': 'rhel_hp-dl380-gen9-5'}})
skipping: [sat63-rhel7] => (item={'key': 'swap', 'value': {'size_g': '15.69', 'vg': 'rhel_hp-dl380-gen9-5'}}) 

TASK [partition-disk : Resize the root filesystem] ***************************************************************************************************************************************************************************************************************************************
ok: [sat63-rhel7] => (item={'key': 'root', 'value': {'size_g': '262.68', 'vg': 'rhel_hp-dl380-gen9-5'}})
skipping: [sat63-rhel7] => (item={'key': 'swap', 'value': {'size_g': '15.69', 'vg': 'rhel_hp-dl380-gen9-5'}}) 

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************************
sat63-rhel7                : ok=7    changed=0    unreachable=0    failed=0   
```

Feel, it is better to also test re-run, just to avoid any issues while re-running the playbook on the same host.

chris1984 commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,27 @@
+```
+git clone git@github.com:yourname/ansible-satellite6
+cd ansible-satellite6
+git checkout -b your-branch-name
````
Should we have them add the remote so they can rebase easily and pull in new changes?

kbidarkar replied:
That suggestion is based on this link, https://help.github.com/articles/fork-a-repo/ :)

chris1984 replied:
Ok, I will mark my comment approved then :)

chris1984 reviewed (changes requested):
Looks good, left 1 comment

omaciel commented:
Was talking to @kbidarkar on IRC about the need to make sure that the underlying version of RHEL does not affect the outcome of this task. Like setting up the firewall rules on RHEL 6 x RHEL 7 :)

flatfender commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,27 @@
+=========
+Install and Run [ansible-lint](https://github.com/willthames/ansible-lint) against your code changes.
+
+Also, Install and Run [ansible-review](https://github.com/willthames/ansible-review) against your code changes.
```
We may want to consider conforming to ansible-galaxy init --offline when creating roles as well.

kbidarkar replied:
Yup, will be sending out a mail about all the best-practices stuff soon.

kbidarkar commented:
@omaciel, ok, filed an issue, here #5 

Wanted to just get started with a role and dir_structure.

Just focused on RHEL7 OS in this PR, to begin with, the ansible-satellite6 repository.

kbidarkar commented:
Guys, nothing is fixed in stone, this initial commit is to get started, we can evolve the project depending upon the needs moving forward. 
I read stuff around ansible, looked at various ansible projects and pulled in what appeared good, also will be sending out the best practices I have found around this.

ehelms commented on README.md:
```diff
@@ -0,0 +1,51 @@
+
+This repository will be Python3.6 based.
+
+You need Ansible installed. 
```
Is there a minimum Ansible version?

kbidarkar replied:
Currently, I am testing/working with 2.6, any suggestions as to which versions should we be supporting from? Currently, I think of from 2.5+ . Thoughts ?

kbidarkar replied:
Maybe I should just test with Ansible 2.5 too and set that as the min version.

kbidarkar replied:
This is still pending, will test and report here and then a re-review. Post which we will merge it.

ehelms commented on ansible.cfg:
```diff
@@ -0,0 +1,6 @@
+module_utils = foreman-ansible-modules/module_utils
+roles_path = $PWD/galaxy_roles:$PWD/roles
+inventory = inventory
+retry_files_enabled = False
```
A few others to consider:

```
force_color = 1
stdout_callback = yaml
display_skip_hosts = false
```

kbidarkar replied:
Ok, will take a look at each of those options and will add it up.

ehelms commented on ansible.cfg:
```diff
@@ -0,0 +1,6 @@
+library = foreman-ansible-modules/modules
+module_utils = foreman-ansible-modules/module_utils
+roles_path = $PWD/galaxy_roles:$PWD/roles
+inventory = inventory
```
I'd recommend making an inventories/ directory and putting the sample in there.

kbidarkar replied:
I like it too, even @flatfender was suggesting the same, I just followed this from Ansible and so kept it at top-level: 
https://docs.ansible.com/ansible/latest/user_guide/playbooks_best_practices.html#directory-layout

kbidarkar replied:
ok, I will make this as `inventory_files/inventory`, also would move `inventory.sample` file to `inventory_files`, else suggest me a name :)

ehelms commented on roles/partition-disk/handlers/main.yml:
```diff
@@ -0,0 +1,2 @@
+---
+# handlers file for partition-disk
```
If these files are gonn abe empty, I'd just delete them to reduce the noise

kbidarkar replied:
In the Best-practices etherpad which I created and shared with @flatfender, I added this point to delete unused files created after we run `ansible-galaxy init`, but @flatfender  mentioned we maintain the dir_structure as it is. 

I think the idea of @flatfender was to have these as it is,  for the first/initial role, so that it be an example/reference for the directory structure we want to put in place.

Maybe moving forward we can remove it, ok will discuss with @flatfender  about it.

ehelms replied:
Sounds good. I tend to think the docs on roles should suffice and the code repository should have as little noise as possible.

flatfender replied:
Yes the idea was consistency around ansible-galaxy init, so that all roles have consistent structure and empty dirs don't hurt, and it makes it more galaxy friendly if we then commit back to galaxy. Once you delete them, then people tend to make things up later and we get inconsistency.

ehelms replied:
I think given that Ansible roles have a defined structure there is no worry about things getting made up. But I do feel "noise" is a valid concern. I dislike going into code bases finding empty useless files and wasting my time on them personally. That makes debugging harder IMO.

ehelms commented on roles/partition-disk/meta/main.yml:
```diff
@@ -0,0 +1,50 @@
+  # - GPLv2
+  # - GPLv3
+  # - Apache
+  # - CC-BY
```
You could get rid of all these comments in here.

ehelms commented on roles/partition-disk/meta/main.yml:
```diff
@@ -0,0 +1,50 @@
+galaxy_info:
+  author: Satellite QE Team
+  description: Satellite QE Team
+  company: RedHat
```
Red Hat

kbidarkar replied:
Argh! I knew about this, my bad will fix it.

ehelms commented on roles/partition-disk/tasks/extend_root_partition.yaml:
```diff
@@ -0,0 +1,16 @@
+      fstype: xfs
+      resizefs: yes
+    with_dict: "{{ ansible_lvm.lvs }}"
+...
```
What are the three dots here?

kbidarkar replied:
I use that for YAML files to mark end of the file, as I had read about it here, https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html#yaml-basics

ehelms commented on roles/partition-disk/tasks/extend_root_partition.yaml:
```diff
@@ -0,0 +1,16 @@
+  - name: "Extend root partition LV to all available VG space"
```
These should be indented at column 0

ehelms commented on roles/partition-disk/tasks/remove_home_partition.yaml:
```diff
@@ -0,0 +1,18 @@
+      state: absent
+      force: yes
+    with_dict: "{{ ansible_lvm.lvs }}"
+...
```
I think these can be dropped

kbidarkar replied:
I use that for YAML files to mark end of the file, as I had read about it here, https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html#yaml-basics  :)

ehelms replied:
I learned something new today. I have never seen that before so I might recommend not including them to reduce confusion. I do a see a mixed use of `---` at the start of YAML across all YAML I have run across.

ehelms commented on roles/partition-disk/tasks/remove_home_partition.yaml:
```diff
@@ -0,0 +1,18 @@
+    mount:
+      name: /home
+      state: absent
+  - name: "Remove /home Logical Volume"
```
I find it helps readability to separate tasks by an empty line just like you would a function definition in python code

kbidarkar committed:
Updated as per comments.

*) Fixed as per yamllint
*) Fixed as per ansible-review and ansible-lint
*) Moved inventory file to inventory_files
*) Added standards.py under rules/ansible-review.
*) Updated ansible.cfg as per suggestions
*) Added some best practices to CONTRIBUTING.md

Signed-off-by: Kedar Bidarkar <kbidarka@redhat.com>
